### ISSUE TITLE
How to replicate paper results?

### ISSUE THREAD

**Issue by artemisp** [mdswyz/DiCMoR#5]
Attempting to retrain the model with MR=0.0 results to significantly lower performance (F1 .55, compared to .85 in the paper). 

Moreover, even loading the pretrained model with MR=0.0 yields about ~5 points lower F1, and similarly, as MR increases the performance gap between reported scores and replicated scores increases up to 20 points in F1 for MR=.7. 

Could you please provide some guidance on what could be the issue? 

Thank you!

**Comment by yamseyoung** [comment]
I got the same result (acc2=.58) as seen in the generated result.csv file, but I noticed that in the train log, the final performance on test data is as the following:
`- MMSA [INFO] - TEST-(dicmor) >>  Acc_2: 0.8155  F1_score: 0.8148  Acc_7: 0.4052  MAE: 0.8233  Loss: 0.8227 `

There seems to be a discrepency in these two results. Could the authors kindly explains this?

**Comment by chenjjcccc** [comment]
> Attempting to retrain the model with MR=0.0 results to significantly lower performance (F1 .55, compared to .85 in the paper).
> 
> Moreover, even loading the pretrained model with MR=0.0 yields about ~5 points lower F1, and similarly, as MR increases the performance gap between reported scores and replicated scores increases up to 20 points in F1 for MR=.7.
> 
> Could you please provide some guidance on what could be the issue?
> 
> Thank you!

May I ask if you've solved it?

**Comment by mdswyz** [comment]
Hi! If you want to obtain a pretrained model, i.e., MR=0.0, you need to remove all codes that are used for missing modality recovery, turn on the gradient of BERT, and set the learning rate to 0.0001. For the F1 .55, I don't know what's going on. It's weird.

**Comment by chenjjcccc** [comment]
> Hi! If you want to obtain a pretrained model, i.e., MR=0.0, you need to remove all codes that are used for missing modality recovery, turn on the gradient of BERT, and set the learning rate to 0.0001. For the F1 .55, I don't know what's going on. It's weird.

Yes, I trained the model according to the instructions in the README, and set MR = 0.5. However, the accuracy is much lower than that in the paper. May I ask if there is something wrong with my operations?

**Comment by mdswyz** [comment]
> > Hi! If you want to obtain a pretrained model, i.e., MR=0.0, you need to remove all codes that are used for missing modality recovery, turn on the gradient of BERT, and set the learning rate to 0.0001. For the F1 .55, I don't know what's going on. It's weird.
> 
> Yes, I trained the model according to the instructions in the README, and set MR = 0.5. However, the accuracy is much lower than that in the paper. May I ask if there is something wrong with my operations?

Here are some suggestions and tricks. You can increase the learning rate when the MR increases (0.001 or 0.002). You can also select the best results by early-stop. Grid-search has also been suggested to improve performance.
